--- jquery.mobile-1.0b1.js	2011-07-03 10:31:49.000000000 -0700
+++ jquery.mobile-1.0b1-hm.js	2011-07-03 10:29:54.000000000 -0700
@@ -1057,72 +1057,78 @@
 					if ( touching ) {
 						triggerCustomEvent( thisObject, "taphold", event );
 					}
 				}, 750 );
 			});
 	}
 };
 
 // also handles swipeleft, swiperight
 $.event.special.swipe = {
+	scrollSupressionThreshold: 10, // More than this horizontal displacement, and we will suppress scrolling.
+	
+	durationThreshold: 1000, // More time than this, and it isn't a swipe.
+	
+	horizontalDistanceThreshold: 30,  // Swipe horizontal displacement must be more than this.
+	
+	verticalDistanceThreshold: 75,  // Swipe vertical displacement must be less than this.
+
 	setup: function() {
 		var thisObject = this,
 			$this = $( thisObject );
-		
-		$this
-			.bind( touchStartEvent, function( event ) {
+
+		$this.bind( touchStartEvent, function( event ) {
+			var data = event.originalEvent.touches ?
+								event.originalEvent.touches[ 0 ] : event,
+				start = {
+					time: ( new Date() ).getTime(),
+					coords: [ data.pageX, data.pageY ],
+					origin: $( event.target )
+				},
+				stop;
+
+			function moveHandler( event ) {
+
+				if ( !start ) {
+					return;
+				}
+
 				var data = event.originalEvent.touches ?
-						event.originalEvent.touches[ 0 ] :
-						event,
-					start = {
-						time: (new Date).getTime(),
-						coords: [ data.pageX, data.pageY ],
-						origin: $( event.target )
-					},
-					stop;
-				
-				function moveHandler( event ) {
-					if ( !start ) {
-						return;
-					}
-					
-					var data = event.originalEvent.touches ?
-							event.originalEvent.touches[ 0 ] :
-							event;
-					stop = {
-							time: (new Date).getTime(),
-							coords: [ data.pageX, data.pageY ]
-					};
-					
-					// prevent scrolling
-					if ( Math.abs( start.coords[0] - stop.coords[0] ) > 10 ) {
-						event.preventDefault();
-					}
+						event.originalEvent.touches[ 0 ] : event;
+
+				stop = {
+					time: ( new Date() ).getTime(),
+					coords: [ data.pageX, data.pageY ]
+				};
+
+				// prevent scrolling
+				if ( Math.abs( start.coords[ 0 ] - stop.coords[ 0 ] ) > $.event.special.swipe.scrollSupressionThreshold ) {
+					event.preventDefault();
 				}
-				
-				$this
-					.bind( touchMoveEvent, moveHandler )
-					.one( touchStopEvent, function( event ) {
-						$this.unbind( touchMoveEvent, moveHandler );
-						if ( start && stop ) {
-							if ( stop.time - start.time < 1000 && 
-									Math.abs( start.coords[0] - stop.coords[0]) > 30 &&
-									Math.abs( start.coords[1] - stop.coords[1]) < 75 ) {
-								start.origin
-								.trigger( "swipe" )
+			}
 
-								.trigger( start.coords[0] > stop.coords[0] ? "swipeleft" : "swiperight" );
-							}
+			$this.bind( touchMoveEvent, moveHandler )
+				.one( touchStopEvent, function( event ) {
+					$this.unbind( touchMoveEvent, moveHandler );
+
+					if ( start && stop ) {
+						if ( stop.time - start.time < $.event.special.swipe.durationThreshold &&
+								Math.abs( start.coords[ 0 ] - stop.coords[ 0 ] ) > $.event.special.swipe.horizontalDistanceThreshold &&
+								Math.abs( start.coords[ 1 ] - stop.coords[ 1 ] ) < $.event.special.swipe.verticalDistanceThreshold ) {
+
+							start.origin.trigger( "swipe" )
+								.trigger( start.coords[0] > stop.coords[ 0 ] ? "swipeleft" : "swiperight" );
 						}
-						start = stop = undefined;
-					});
-			});
+					}
+					start = stop = undefined;
+				});
+		});
 	}
 };
 
 (function($){
 	// "Cowboy" Ben Alman
 	
 	var win = $(window),
 		special_event,
 		get_orientation,
 		last_orientation;
